pkg/asset/installconfig/azure/validation.go

```diff
@@ -69,3 +72,44 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
 	}
 	return field.ErrorList{field.Invalid(fldPath, subnetName, fmt.Sprintf("subnet %s address prefix is outside of the specified machine networks", ip))}
 }
+
+// validateRegion checks that the desired region is valid and available to the user
+func validateRegion(client API, fieldPath *field.Path, p *aztypes.Platform) field.ErrorList {
+	locations, err := client.ListLocations(context.TODO())
+	if err != nil {
+		return field.ErrorList{field.Invalid(fieldPath, p.Region, "failed to retrieve available regions")}
+	}
+
+	availableRegions := map[string]string{}
+	for _, location := range *locations {
+		availableRegions[to.String(location.Name)] = to.String(location.DisplayName)
+	}
+
+	if _, ok := availableRegions[p.Region]; !ok {
+		errMsg := fmt.Sprintf("region %q is not valid or not available for this account", p.Region)
+
+		lowercaseRegion := strings.ToLower(p.Region)
```

Comment:
https://github.com/terraform-providers/terraform-provider-azurerm/blob/3e8852744df3d18f4d064cc9adf39ea131e96504/azurerm/internal/location/normalize.go#L5-L19

Reply:
That's an internal package so it can't be imported, but the logic is simple so I reproduced it here.